src/fork_gc.c

```diff
@@ -795,48 +911,38 @@ static bool FGC_parentHandleTags(ForkGC *gc, int *ret_val, RedisModuleCtx *rctx)
     if (sctx) {
       SearchCtx_Free(sctx);
     }
-    if (keyName) {
-      RedisModule_FreeString(rctx, keyName);
-    }
     if (idxKey) {
       RedisModule_CloseKey(idxKey);
     }
     if (hasLock) {
       FGC_unlock(gc, rctx);
       hasLock = 0;
     }
-    if (shouldReturn) {
+    if (status != FGC_COLLECTED) {
       freeInvIdx(&idxbufs, &info);
-      rm_free(fieldName);
-      return false;
     } else {
       rm_free(idxbufs.changedBlocks);
     }
   }
 
   rm_free(fieldName);
-  return true;
+  return status;
 }
 
-void FGC_parentHandleFromChild(ForkGC *gc, int *ret_val) {
-  while (FGC_parentHandleTerms(gc, ret_val, gc->ctx))
-    ;
-
-  if (!(*ret_val)) {
-    goto done;
-  }
-
-  while (FGC_parentHandleNumeric(gc, ret_val, gc->ctx))
-    ;
+int FGC_parentHandleFromChild(ForkGC *gc) {
+  FGCError status = FGC_COLLECTED;
 
-  if (!(*ret_val)) {
-    goto done;
+#define COLLECT_FROM_CHILD(e)               \
+  while ((status = (e)) == FGC_COLLECTED) { \
```

Comment:
Whats wrong with just ';'?

Reply:
I find that `{}` makes it easier to see there is no loop body.. also clang-format has weird indentation for `;` as a loop body